Add float_classify function to float library

Timmmm opened the pull request:
This is a standard operation in IEEE 754 floats - see `class()` in the "General operations" section of the spec.

It's also implemented as the `fclass` instruction in RISC-V.

Timmmm commented:
Originally from [here](https://github.com/riscv/sail-riscv/pull/1288) but I think it probably makes more sense to put it in with the rest of these float functions.

jordancarlin reviewed (commented):
You should add tests in `test/float/classify.sail` like all of the other floating point library functions. Otherwise, LGTM.

Commit:
Add float_classify function to float library

This is a standard operation in IEEE 754 floats - see `class()` in the "General operations" section of the spec.

It's also implemented as the `fclass` instruction in RISC-V.

Commit:
Rename denormal to subnormal

I didn't know the difference between these so I looked into it. "Denormal" is an obsolete from ancient floating point formats that had an explicit leading 1 in the significand. The only example I could find is the IBM 709 from the 50s! For those formats it was possible to have multiple representations of a single number, e.g. 10 could be 2^3 * 0b1.01000 (normal) but also 2^4 * 0b0.10100, 2^5 * 0b0.00101, etc. which are denormal (but not subnormal because the exponent is not the minimum value).

Since that is impossible in all non-ancient formats, there are no denormals that are not also subnormals and all modern documents use the term "subnormal" instead, including the RISC-V spec and the the IEEE 754 spec (since 2008).

Timmmm commented:
Rebased & test added. I verified it passes manually.
